### PR TITLE
Add semantic patterns documentation

### PR DESCRIPTION
**Pull request by youknowriad**
related #28737 

While I try to work on #28737 I noticed that we actually already support these kind of semantic patterns but that it was not documented properly.

It was not easy to come with the right wording here. So please happy to take on suggestions.

### CODE REVIEW

File: docs/reference-guides/block-api/block-patterns.md [hunk 1]
```diff
@@ -135,4 +136,48 @@ function my_plugin_unregister_my_pattern_categories() {
 }
  
 add_action( 'init', 'my_plugin_unregister_my_pattern_categories' );
-```
\ No newline at end of file
+```
+
+## Block patterns contextual to block types
+
+It is possible to attach a block pattern to one or multiple block types. This makes the block pattern as an available transform for that block type.
```
Comment (ryanwelcher) [inline docs/reference-guides/block-api/block-patterns.md]:
```suggestion
It is possible to attach a block pattern to one or more block types. This adds the block pattern as an available transform for that block type.
```

**Review by ryanwelcher — APPROVED** [review]
Just added a small change

**Review by gziolo — APPROVED** [review]
Looks great! It needs to be better advertised . There is also something with attributes where you can mark their role as `content` in case you want to inject that into the pattern. @ntsekouras might know better if that applies here.

**Comment by youknowriad** [comment]
Good call @gziolo Yeah, the role is indeed related. It's still experimental though, so let's wait a bit before adding that information.

File: docs/reference-guides/block-api/block-patterns.md [hunk 2]
```diff
@@ -135,4 +136,48 @@ function my_plugin_unregister_my_pattern_categories() {
...
+
+```php
+<?php
+register_block_pattern(
+	'my-plugin/my-header',
+	array(
+		'title'      => __( 'My Header', 'my-plugin' ),
+		'categories' => array( 'header' ),
+		// Assigning the pattern the "header" area.
+		'blockTypes' => array( 'core/template-part/header' ),
```
Comment (ntsekouras) [inline docs/reference-guides/block-api/block-patterns.md:179]: It's worth noting that `core/template-part/header` is something that is used only by `template parts` and is not a generic API for block variations. It had been discussed to support block variations with the same format, but eventually left for template parts to override the filtering patterns function with that naming convention.

**Comment by ntsekouras** [comment]
I created a follow up for this one here: https://github.com/WordPress/gutenberg/pull/38809, if you think it brings some value 😄

**Pull request merged**